packages/amplify-codegen-appsync-model-plugin/src/utils/process-connections.ts

```diff
@@ -181,12 +182,24 @@ export function processConnections(
             license: License;
           }
           */
-          return {
-            kind: CodeGenConnectionType.HAS_ONE,
-            associatedWith: otherSideField,
-            connectedModel: otherSide,
-            isConnectingFieldAutoCreated,
-          };
+          switch (connectionFields.length) {
+            case 0:
+              return {
+                kind: CodeGenConnectionType.HAS_ONE,
+                associatedWith: otherSideField,
+                connectedModel: otherSide,
+                isConnectingFieldAutoCreated,
+              };
+            case 1:
+              return {
+                kind: CodeGenConnectionType.BELONGS_TO,
+                connectedModel: otherSide,
+                isConnectingFieldAutoCreated,
+                targetName: connectionFields[0] || makeConnectionAttributeName(model.name, field.name),
+              };
+            default:
+              throw new Error('DataStore only support one key in field');
+          }
```

Comment:
The check should be based on if the field is nullable or non-nullable. When an field is non-nullable, it beconmes belogns to and when it is nullable it becomes `hasOne`

Reply:
yes , thats true, But What about the case when @connection directive has a field which is non nullable. for example 
```graphql
          type Person @model {
            licenseId: ID!
            license: License @connection(fields: ["licenseId"])
          }
          type License @model {
            id: ID!
            holder: Person! @connection
          }
```
Here licence Id is non nullable . Wont this mean every licence(LicenceID) belongs to a person?

Reply:
Logically doesn't it make sense. For a license to exists, there needs to be a person. So a person has a license and a license belongs to a person